src/ipfs/basic.ts

```diff
@@ -105,6 +105,8 @@ export const size = async (cid: CID): Promise<number> => {
 }
 
 export const attemptPin = async (cid: CID): Promise<void> => {
+  if (CID.isCID(cid)) throw new Error("Expected a CID object")
```

Comment:
You forgot the negation, but otherwise all good! 😄 ie. you should check if it's **not** a CID.

Reply:
Oh. I was testing the commit signing here. I'm not done. I had a rethink of the implementation and I will be pushing that now. We might need to hop on a call and figure how the check should be done based on the discussion we had yesterday.